frontend/components/MainPage/header/AgentButton.tsx

```diff
@@ -205,21 +218,39 @@ const AgentNotRunningButton = () => {
       showNotification?.('Error while showing "running" notification');
     }
 
+    // TODO: remove this workaround, middleware should respond when agent is staked & confirmed running after `createService` call
+    await new Promise((resolve) => setTimeout(resolve, 5000));
+
     // Can assume successful deployment
-    // resume polling, optimistically update service status (poll will update, if needed)
-    setIsServicePollingPaused(false);
-    setIsBalancePollingPaused(false);
     setServiceStatus(DeploymentStatus.DEPLOYED);
+
+    // update provider states sequentially
+    try {
+      await updateServicesState(); // reload the available services
```

Comment:
if the state will be updated here anyway, does it still make sense to do `setServiceStatus(DeploymentStatus.DEPLOYED);` in the 225 line?

Reply:
good point.. 
the set state was supposed to go before the timeout, will update!